File: Pendu.js
const dictionary = ["Chocolat","Science","Informatique","Musique","Geologie","Gastronomie"]
let find = false
let userSelect
// let regexInput = new RegExp("/^[a-zA-Z]{1}$")

function getOneWord(table) {
    
    const myWord = dictionary[Math.floor(Math.random() * table.length)]
    const myArray = myWord.split("")
    for(let i = 0; i < myArray.length; i++) {
        myArray[i] = myArray[i].toUpperCase()
    }
    return myArray
}

function buildUserArray(wordArray) {
    let ArrayUser = []
    for(let i = 0; i < wordArray.length; i++) {
        ArrayUser.push("X")
    }
    return ArrayUser
}

function letsgame() {
    const myWordSplit = getOneWord(dictionary)
    console.log(myWordSplit)
    let ArrayUser = buildUserArray(myWordSplit)
    while(find == false) {
        if(myWordSplit == ArrayUser) {
            find = true
            console.log("the winner is !")
        }
        console.log("choose a letter")
        let userInput = prompt().toUpperCase()
        while(userInput.length > 1) {
            console.log("Nous vous demandons une seule lettre")
            userInput = prompt().toUpperCase()
        }
        console.log("Vous avez choisie la lettre : " + userInput)
        for(let i = 0; i < ArrayUser.length; i++) {
            if(userInput == myWordSplit[i]) {
                console.log("trouvé !")
                ArrayUser[i] = userInput
            }
        }

        console.log(ArrayUser)
    }
    console.log(" Vous avez gagné !")
    console.log(" Voulez-vous recommencer ? Y | N")
    userSelect = prompt()
}

letsgame()

switch (userSelect) {
    case 'Y' :
        letsgame()
    case "N" :
        break
    default :
    console.log('Wrong entry please try again')
        userSelect = prompt()
}

File: Pendu.pl
#!usr/bin/perl
my @dictionary = ("Chocolat","Science","Informatique","Musique","Geologie","Gastronomie");
my $victory = 0;
my $userSelect

sub getOneWord(tableau) {
    my $randNbr = rand(length(tableau))
    my $myWord = tableau[$randNbr]
    my @myWordSplit.split("", $myWord)

    for( my $i = 0; $i < length(@myWordSplit) ; $i++) {
        @myWordSplit[$i] = @myWordSplit[$i].uc()
    }
    return @myWordSplit
}

sub buildUserArray(wordArray) {
    my @ArrayUser = []
    for(my $i = 0; i < length(wordArray); $i++) {
        push @ArrayUser, 'X'
    }
    print "@ArrayUser"
}


#Tableau et Tableau associatif

my @tab1 = ("hello", 3, $a); #tableau simple
my %tab2 = (
    "insert1" => "hello"
    "insert2" => 3
)
my @tabMap = (2,3,4,5,6)
my @tabMapSort = map( { $_* = 2} @tabMap) #multiplie tous les elements de @tapMap par deux

#les conditions if, while,  

if ($a == 1) {
    print "Je suis egal"
} else {
    print "Je ne suis pas égal"
}

print "Comparaison ok" if(defined($a))

while ($a =< 10) {
    print "Encore un !"
    $a++
}




